src/hotspot/share/gc/g1/g1HeapSizingPolicy.cpp

```diff
@@ -198,6 +198,14 @@ size_t G1HeapSizingPolicy::young_collection_expansion_amount() {
 }
 
 static size_t target_heap_capacity(size_t used_bytes, uintx free_ratio) {
+  assert(free_ratio <= 100, "precondition");
```

Comment:
Doesn't debug.hpp have precond for this?

Reply:
It does, and hardly anyone uses it.